homeassistant/components/rainbird/config_flow.py

```diff
@@ -125,8 +125,9 @@ async def async_finish(
         options: dict[str, Any],
     ) -> FlowResult:
         """Create the config entry."""
-        await self.async_set_unique_id(serial_number)
-        self._abort_if_unique_id_configured()
+        if serial_number:
```

Comment:
Looking at this and the attached issue it seems it would be better to remove unique id completely and use `_async_abort_entries_match()` to see if the entry options are matching.

Reply:
Hi, I may need some additional help to understand. Can you elaborate?

One thing i'd like to clarify is that most devices do have serial numbers, but some devices appear to return a zero. My intent is to keep a unique id for devices that do have a unique id, and not have one for devices that don't have one.

Happy to hear your thoughts in more detail.

Reply:
Maybe I want both? Use `_async_abort_entries_match` when there is no serial? Sounds like that gives some of the benefits of checking if a unique id already exists (same host:port/password) when there is no unique id?

Reply:
I was just thinking that since you need to use `_async_abort_entries_match()` for testing when there is no serial why not always use that instead of mixing it together with unique_id.

Reply:
Serial matching is always best, so using the unique ID would be nice and abort on that if possible. The `_async_abort_entries_match` is nice, but not as robust as working with a unique ID. Therefore, I think the suggestion above by @allenporter makes a lot of sense.